src/facades/fspiop.ts

```diff
@@ -109,10 +122,15 @@ export const FspiopTransformFacade = {
       if (!TypeGuards.FSPIOP.quotes.put.isSource(source)) {
         throw new Error('Invalid source object for put quotes');
       }
+      if (!isTestingMode && !source.$context) {
+        throw new Error('Invalid source object for put quotes, missing $context');
+      }
```

Comment:
The need for a "testing mode"check is not quite clear to me. If you could shed more light, it might be possible to achieve what the goal is without going this route.

Reply:
Here and throughout.

Yes. It's so we never have a default value end up in a production request.
The visibility of how a request gets transformed is opaque. 
There aren't many cases where I think this could happen but it's better to be explicit.